web-console/knikubevirt/Affinity/affinity.md

```diff
@@ -0,0 +1,61 @@
+
+## creating an affinity rule from Affinity Rules page
+
+Clicking on the "Add affinity rule" button at the top of the affinity list will open up the "Add affinty rule" modal.
+
+![Affinity rule modal - default](img/D2-1-0.jpg)
+Going top to bottom, the user will choose to create an affinity (attachment), which is pre-selected, or change to Anti-affinity (rejection).
+Then, the user will select the resources he wishes to apply that rule to by adding their representative labels to the "Select resources by label" field.
+
+*Please note that we are aware that it is quite dificult to select resources via labels alone, without seeing any mention on the resources those labels represent, without selecting resources directly first or without any label autocomplete. We already have a few designs to resolve that issue, but these designs affect the rest of the system too and are out of scope. for the moment, we are aiming to get the most minimal design merged and take it from there, while sticking to how it actually works in k8.*
+
+![Force rule](img/D2-1-1.jpg)
+Before applying, the user can check the "Force this rule" checkbox, making this rule a required condition when scheduling the resources targeted by the added labels above.
+![applying rule](img/D2-1-2.jpg)
+The user chose to apply this rule without "forcing" it.
+The user now returns to the Affinity rules list.
+
+# Affinity rules list on a resource details page
+
+![VM details page - overview tab - affinity section](img/D1-0-0.jpg)
+Affinity rules can be viewed and managed per resource as well. The list is located in the resource details page, on the overview tab in Affinity rules section under 'Services' if those exist.
+
+## Creating an affinity rule from a resource details page
+
+![Affinity rule modal - default](img/D1-1-0.jpg)
+
+Clicking on the "Add affinity rule" button at the top of the affinity list will open up the "Add affinty rule" modal.
+It is the same as the modal from the Affinity Rules list page except one thing: The resource itself is already added to the "Select resources by label" field.
+The resource will be mentioned by it's name. 
+Technically, this "direct selection" of this resource is not actually a direct selection. We already mentioned that selecting a resource directly is not the best way since that resource will crush and regenerate more than a few times. Instead, that  selection will provide a new, uniquew label to the specified resource. We do not need to surface this to the user, at least not for now.
+
+![Affinity rule modal - selecting labels](img/D1-1-1.jpg)
+User enters a label. The resource name remains first in the field, even after adding labels. It will not be viewed as a label, but as a placeholder text.
+The rule is ready to be applied. The user clicks 'Apply' and returns to the resource details page, where he left off.
+
+
+## Bypassing a resource scheduling failure due to a forced affinity rule
+
+![Resource scheduling failure - modal](img/D3-1-0.jpg)
```

Comment:
We should brainstorm (in a separate PR :) ) how the user might get notified of these scheduling failures if they don't actually attempt to create this resource manually. Some resources can be a part of a larger deployment or driven via a script outside of the UI and it would be great to maybe work in how these warnings might surface say in the dashboard.

Reply:
+1 
I excluded the entry point to this "Schedule failure" modal since I'm not sure if and how we push these kinds of notifications to the user. @lizsurette  Should I create an issue for that?

Reply:
Yes for sure. No need to hold this PR up on that discussion.

Reply:
Dropping a link to the '[Resource scheduling failure](https://github.com/openshift/openshift-origin-design/issues/234)' issue here